modelconverter/utils/onnx_tools.py

```diff
@@ -176,3 +181,975 @@ def onnx_attach_normalization_to_inputs(
+                    raise ValueError(
+                        f"Split and Concat axis mismatch: {split_node.attrs['axis']} != {concat_node.attrs['axis']}"
+                    )
+
+                channels_axis = split_node.attrs["axis"]
+
+                for inter_node in intermediate_nodes[:-1]:
+                    constant = self.get_constant_value(
+                        inter_node, self.get_constant_map(self.onnx_gs)
+                    )
+                    if constant is None:
+                        continue
+                    constant_value, constant_idx = constant
+                    if constant_value.ndim == 1:
+                        continue
+
+                    if (
+                        constant_value.shape[channels_axis]
+                        != conv_weights.values.shape[1]
+                    ):
+                        raise ValueError(
+                            f"Spatial dimensions mismatch between Conv and intermediate node {inter_node.name}: {constant_value.shape[channels_axis]} != {conv_weights.values.shape[1]}"
+                        )
+
+                    # Reverse the order of the constant tensor channels
+                    inter_node.inputs[constant_idx].values = np.flip(
+                        constant_value, axis=channels_axis
+                    )
+
+                # Reverse the order of the channels of the conv node at the axis
+                conv_weights.values = np.flip(
+                    conv_weights.values, axis=channels_axis
+                )
+
+                nodes_to_remove.append(split_node)
+                nodes_to_remove.append(concat_node)
+
+                connections_to_fix.append(
+                    (
+                        concat_node.outputs[0],
+                        split_node.inputs[0],
+                    )
+                )
+
+                break
+
+        self.graph_cleanup([], nodes_to_remove, connections_to_fix)
+        self.onnx_model = gs.export_onnx(self.onnx_gs)
+
+        self.optimize_onnx()
+
+    def modify_onnx(self) -> None:
+        """Modify the ONNX model by applying a series of optimizations.
+
+        @param passes: List of optimization passes to apply to the ONNX model
+        @type passes: Optional[List[str]]
+        """
+
+        logger.debug("Substituting Div -> Mul nodes...")
+        self.substitute_node_by_type(source_node="Div", target_node="Mul")
+        logger.debug("Substituting Sub -> Add nodes...")
+        self.substitute_node_by_type(source_node="Sub", target_node="Add")
+        logger.debug(
+            "Fusing Add and Mul nodes to BatchNormalization nodes and then into Conv nodes..."
+        )
+        self.fuse_add_mul_to_bn()
+        logger.debug("Fusing Add and Mul nodes to Conv nodes...")
+        self.fuse_comb_add_mul_to_conv()
+        self.fuse_single_add_mul_to_conv()
+        logger.debug("Fusing Split and Concat nodes to Conv nodes...")
+        self.fuse_split_concat_to_conv()
```

Comment:
I think It would be beneficial to run `compare_outputs` function after every modification is done. And have the logger / converter give a big red warning to which stage the model was modified to. This would also help with faster debugging for internal use.

Reply:
+1, easier potential debugging outways the small addition in time needed for conversion IMO